packages/adapters-library/src/adapters/beefy/products/moo-token/beefyMooTokenAdapter.ts

```diff
@@ -111,87 +97,81 @@ export class BeefyMooTokenAdapter implements IProtocolAdapter {
-            productId: this.productId,
-            vaultId: vault.id,
-            platformId: vault.platformId,
-            vaultAddress: vault.earnedTokenAddress,
-            poolAddress: vault.tokenAddress,
-            strategyTypeId: vault.strategyTypeId,
-            chain,
-          },
-          'Protocol type not found',
-        )
-        return undefined
-      }
 
-      // test that we can indeed fetch the breakdown, otherwise we don't include the vault
-      // in the list
-      const [protocolToken, underlyingToken, breakdown] = await Promise.all([
-        this.helpers.getTokenMetadata(vault.earnedTokenAddress),
-        this.helpers.getTokenMetadata(vault.tokenAddress),
-        breakdownFetcherMap[protocolType](
-          {
-            protocolTokenAddress: vault.earnedTokenAddress,
-            underlyingLPTokenAddress: vault.tokenAddress,
-            blockSpec: { blockTag: undefined },
-          },
-          this.provider,
-        ),
-      ])
-
-      const breakdownTokenMetadata = await Promise.all(
-        breakdown.balances.map((balance) =>
-          this.helpers.getTokenMetadata(balance.tokenAddress),
-        ),
-      )
-
-      return {
-        ...protocolToken,
-        underlyingTokens: breakdownTokenMetadata,
-        underlyingLPToken: underlyingToken,
-        unwrapType: protocolType,
+      try {
+        const [protocolToken, underlyingToken] = await Promise.all([
+          this.helpers.getTokenMetadata(vault.earnedTokenAddress),
+          this.helpers.getTokenMetadata(vault.tokenAddress),
+        ])
+
+        return {
+          ...protocolToken,
+          underlyingTokens: [underlyingToken],
+        }
+      } catch (error) {
+        return
       }
     })
   }
 
   async unwrap({
-    protocolTokenAddress,
     blockNumber,
+    protocolTokenAddress,
   }: UnwrapInput): Promise<UnwrapExchangeRate> {
-    const {
-      underlyingTokens,
-      unwrapType,
-      underlyingLPToken,
-      ...protocolToken
-    } = await this.getProtocolTokenByAddress(protocolTokenAddress)
+    const protocolTokenMetadata =
+      await this.getProtocolTokenByAddress(protocolTokenAddress)
+
+    const underlyingTokenConversionRate = await this.unwrapProtocolToken(
+      protocolTokenMetadata,
+      blockNumber,
+    )
 
     return {
-      ...protocolToken,
+      ...protocolTokenMetadata,
       baseRate: 1,
-      type: TokenType['Protocol'],
-      tokens: [
-        {
-          ...underlyingLPToken,
-          underlyingRateRaw: BigInt(10 ** underlyingLPToken.decimals),
-          type: TokenType['Underlying'],
-        },
-      ],
+      type: TokenType.Protocol,
+      tokens: underlyingTokenConversionRate,
     }
   }
 
+  protected async unwrapProtocolToken(
+    protocolTokenMetadata: Erc20Metadata,
+    blockNumber?: number | undefined,
+  ): Promise<UnwrappedTokenExchangeRate[]> {
+    const [underlyingToken] = await this.getUnderlyingTokens(
+      protocolTokenMetadata.address,
+    )
```

Comment:
```suggestion
    const { underlyingTokens: [underlyingToken] } =
      await this.getProtocolTokenByAddress(protocolTokenAddress)
```

It's almost the same amount of code as calling `getUnderlyingTokens`, which is also only used once.